examples/integrations/langchain4j/coffee-shop-assistant-se/src/main/java/io/helidon/examples/integrations/langchain4j/se/coffee/shop/assistant/ai/MetricsChatModelListener.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.examples.integrations.langchain4j.se.coffee.shop.assistant.ai;
+
+import java.util.concurrent.TimeUnit;
+
+import io.helidon.metrics.api.DistributionStatisticsConfig;
+import io.helidon.metrics.api.DistributionSummary;
+import io.helidon.metrics.api.MeterRegistry;
+import io.helidon.metrics.api.Metrics;
+import io.helidon.metrics.api.Tag;
+
+import dev.langchain4j.model.chat.listener.ChatModelErrorContext;
+import dev.langchain4j.model.chat.listener.ChatModelListener;
+import dev.langchain4j.model.chat.listener.ChatModelRequest;
+import dev.langchain4j.model.chat.listener.ChatModelRequestContext;
+import dev.langchain4j.model.chat.listener.ChatModelResponse;
+import dev.langchain4j.model.chat.listener.ChatModelResponseContext;
+
+/**
+ * Creates metrics that follow the
+ * <a href="https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-metrics/">Semantic Conventions for GenAI Metrics</a>.
+ */
+public class MetricsChatModelListener implements ChatModelListener {
+
+    private static final String GEN_AI_CLIENT_OPERATION_START_TIME = "GEN_AI_CLIENT_OPERATION_START_TIME";
+    private static final String GEN_AI_CLIENT_TOKEN_USAGE_METRICS_NAME = "gen_ai.client.token.usage";
+    private static final String GEN_AI_CLIENT_OPERATION_DURATION_METRICS_NAME = "gen_ai.client.operation.duration";
+
+    private final MeterRegistry meterRegistry;
+    private DistributionSummary clientTokenUsage;
+    private final DistributionSummary.Builder clientTokenUsageBuilder;
+    private DistributionSummary clientOperationDuration;
+    private final DistributionSummary.Builder clientOperationDurationBuilder;
+
+    /**
+     * Constructs a {@code MetricsChatModelListener} instance.
+     */
+    public MetricsChatModelListener() {
+        this.meterRegistry = Metrics.globalRegistry();
+
+        DistributionStatisticsConfig.Builder clientTokenUsageStatisticsConfigBuilder = DistributionStatisticsConfig.builder()
```

Comment:
IIRC from our earlier conversations about this, these bucket boundary values come from a credible source--either common practice or some document. Maybe add a comment linking to that so readers know where the values come from.

Reply:
Have added comment with link to the doc.